website/docs/opengl/c1/3-hello-texture.md

```diff
@@ -0,0 +1,388 @@
+```
+
+You don't need to understand what all these values mean! The OpenGL documentation explains it all in depth. Right now
+let's just continue with the basics!
+
+Now, with the texture data sent off to the GPU, we need to configure the texture parameters. This is one of the most important steps. It is with these
+parameters that the shader knows how to sample colors from the texture using our provided texture coordinates.
+
+For now, you don't need to understand what these do. We will elaborate on it in the future. Just add these lines after you upload the texture data.
+```c#
+_gl.TextureParameter(_texture, TextureParameterName.TextureWrapS, (int) TextureWrapMode.Repeat);
+_gl.TextureParameter(_texture, TextureParameterName.TextureWrapT, (int) TextureWrapMode.Repeat);
+_gl.TextureParameter(_texture, TextureParameterName.TextureMinFilter, (int) TextureMinFilter.Linear);
+_gl.TextureParameter(_texture, TextureParameterName.TextureMagFilter, (int) TextureMagFilter.Linear);
+```
+
+And now, we just need to unbind the texture.
+```c#
+_gl.BindTexture(TextureTarget.Texture2D, 0);
+```
+
+## Handling texture data in shader
+Now, let's finally show it on the screen!
+
+In your fragment shader, add this `uniform` declaration:
+```
+uniform sampler2D uTexture;
+```
+
+And change this line:
+```
+void main()
+{
+//  -out_color = vec4(frag_texCoords.x, frag_texCoords.y, 0, 1.0);
+    out_color = texture(uTexture, frag_texCoords);
+}
+```
+
+In GLSL, you usually dont work with textures directly. The most common way to access a texture is using a `sampler`.
+In our case, we are using `sampler2D` for our 2D texture.
+To read the texture data, the `texture` method is used. It samples the color of the texture at the
+specified texture coordinates and uses the previously set parameters to interpolate that color.
+
+But pay attention! If you have good eyes, you noticed our texture coordinates are between the ranges of `0.0` to `1.0` This is because the `texture` method works
+with normalized values! If you dont understand, think about a 250x500 pixels image. If you want to get the pixel at the position
+(250, 250), we need to send (1, 0.5) as the texture coordinate. You can derive the pixel coordinate from the normalized coordinate by multiplying by the axis' size.
+
+<?# Info "You can use the equation ` 1/size * pixel_position ` to get the normalized coordinate for a particlar axis!" /?>
+
+Now, if you run the program, you will see just a black quad. It's because we need to bind the texture before the draw call!
+If you don't do it, the last applied texture will be used for this mesh, in our case, since we have no texture bound, no texture is used.
+
+To solve it, go to your `OnRender` method and, below the `_gl.UseProgram()` call, first activate the texture slot that you will use:
+```c#
+_gl.ActiveTexture(TextureUnit.Texture0);
+```
+And after, bind the texture!
+```c#
+_gl.BindTexture(TextureTarget.Texture2D, _texture);
+```
```

Comment:
You're never explaining how to set the sampler2D uniform to an actual value!

Reply:
oh, right! Sry, i will explain it better when i rewrite the explain of how feeding and binding textures work!